shark/src/main/java/shark/internal/PathFinder.kt

```diff
@@ -591,6 +591,30 @@ internal class PathFinder(
     }
   }
 
+  /**
+   * Every time we find a new object as we traverse the graph, we call this method.
+   *
+   * This is a twist on the traditional dominator tree / retained size algorithms: dominators here
+   * can only be retained instances, which means we don't need to track that many dominators.
+   *
+   * Here's how it works:
+   *
+   * - If [objectId] has no dominator known yet but has already been added to visit, then we know
+   * we've been here for that object already and we can bail.
```

Comment:
`Bail` is too slang IMO. Replace with some technical equivalent? `Skip`?

Reply:
=> and we did not add it to the map of dominated instances so it can't be dominated any more, we can stop there.

Reply:
Too long for a single sentence IMO.
```
  If [objectId] has no dominator known yet but has already been added to visit, then we know we've been here for that object already. We did not add this object to the map of dominated instances so it can't be dominated any more, we can stop there.
```
?